src/main/java/com/flightstats/hub/dao/aws/MaxItemsEnforcer.java

```diff
@@ -0,0 +1,89 @@
+package com.flightstats.hub.dao.aws;
+
+import com.flightstats.hub.dao.ChannelService;
+import com.flightstats.hub.dao.Dao;
+import com.flightstats.hub.metrics.ActiveTraces;
+import com.flightstats.hub.model.ChannelConfig;
+import com.flightstats.hub.model.ContentKey;
+import com.flightstats.hub.model.DirectionQuery;
+import com.flightstats.hub.util.TimeUtil;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+@Slf4j
+@Singleton
+public class MaxItemsEnforcer {
+
+    private final ContentRetriever contentRetriever;
+    private final ChannelService channelService;
+    private final Dao<ChannelConfig> channelConfigDao;
+
+    @Inject
+    public MaxItemsEnforcer(ContentRetriever contentRetriever,
+                            ChannelService channelService,
+                            @Named("ChannelConfig") Dao<ChannelConfig> channelConfigDao) {
+        this.contentRetriever = contentRetriever;
+        this.channelService = channelService;
+        this.channelConfigDao = channelConfigDao;
+    }
+
+    void updateMaxItems(Iterable<ChannelConfig> configurations) {
+        log.info("iterating channels for max items update");
+        for (ChannelConfig config : configurations) {
+            if (config.getMaxItems() > 0 && !config.getKeepForever()) {
+                updateMaxItems(config);
+            }
+        }
+    }
+
+    @SneakyThrows
+    void updateMaxItems(String channelName) {
+        Optional<ChannelConfig> channelConfig = Optional.ofNullable(channelConfigDao.get(channelName));
```

Comment:
As a stylistic preference, I'd probably chain this with the next line instead of using an intermediate variable:
```Optional.ofNullable(...)
    .filter(....)....
```